CMakeLists.txt

```diff
@@ -0,0 +1,342 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(facil.io VERSION 0.8.0)
+
+# Set the source directories
+set(LIB_ROOT lib)
+set(LIB_CONCAT_FOLDER fio-stl)
+
+include_directories(".")
+
+option(ENABLE_SHARED "Build facil.io as shared library" OFF)
```

Comment:
I suggest using the project name as a prefix for all options — e.g., FACIL_ENABLE_SHARED. That will group all of the options together in `ccmake` and `cmake-gui`. That's useful in general, but it's particularly important for projects like this that are really meant to be included in larger projects.

That aside, on the topic of shared/static, there are other options. One is to not declare the library type. In that case, CMake will try to infer from context. Another is to create two libraries, one shared and one static; then the user can just choose the one they want to link to. Yet another is to make `fio-stl.h` the only source file, and declare this a header-only library (https://dominikberner.ch/cmake-interface-lib/).

I'm only one day into looking at this variant of facil.io, so there's a lot I don't yet know or understand, but from what I see so far it seems like a shared facil library doesn't make much sense anyway; so much of what it does is (or can be) configured at compile time.